internal/pkg/task/config_runner.go

```diff
@@ -36,6 +46,18 @@ func (r *NetworkConfigRunner) Run() ([]*Task, error) {
 		}
 	}
 
+	if r.Subnets == nil {
+		subnets, err := r.VPCGetter.SubnetIDs(ec2.FilterForDefaultVPCSubnets)
```

Comment:
should this be? 
```suggestion
		subnets, err := r.VPCGetter.PublicSubnetIDs()
```

Reply:
The reason why we were asking for only public subnets for `EnvRunner` is because when copilot creates subnets for env, it doesn't include the NAT gateway for private subnets.

I am not sure if we should do that here as well.. It is possible that the user has a default subnet that is private, but have the NAT gateway attached.